examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -0,0 +1,113 @@
+"""
+==================================================
+Balance model complexity and cross-validated score
+==================================================
+
+This example balances model complexity and cross-validated score by
+finding a decent accuracy within 1 standard deviation of the best accuracy
+score while minimising the number of PCA components.
+
+The figure shows the trade-off between cross-validated score and number of
+PCA components. The balanced case is when n_components=6 and accuracy=0.80,
+which falls into the range within 1 standard deviation of the best accuracy
+score.
+"""
+# Author: Wenhao Zhang <wenhaoz@ucla.edu>
+
+print(__doc__)
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.datasets import load_digits
+from sklearn.decomposition import PCA
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import Pipeline
+from sklearn.svm import LinearSVC
+
+
+def lower_bound(cv_results):
+    """
+    Calculate the lower bound within 1 standard deviation
+    of the best `mean_test_scores`.
+
+    Parameters
+    ----------
+    cv_results : dict of numpy(masked) ndarrays
+        See attribute cv_results_ of `GridSearchCV`
+
+    Returns
+    -------
+    float
+        Lower bound within 1 standard deviation of the
+        best `mean_test_score`.
+    """
+    best_score_idx = np.argmax(cv_results['mean_test_score'])
+
+    return (cv_results['mean_test_score'][best_score_idx]
+            - cv_results['std_test_score'][best_score_idx])
+
+
+def refit_callable(cv_results):
```

Comment:
would probably make sense to call this something like `best_low_complexity` or something that actually shows what it does.